### ISSUE TITLE
Conditional revalidation

### ISSUE THREAD

**Issue by vivekimsit** [vercel/swr#337]
This library has been super helpful so far and I like the ease of configuration and get the things up and running.

I have a use case where I am not sure what would be the best way. Taking an example, I have a Car details page where you can track the progress of current status for the order of your car. I am using refresh interval to fetch the details after every 5 mins but I need a way to stop fetching details once the Car is shipped. I am not sure how to do that using the current API.

Thanks,

**Comment by sergiodxa** [comment]
You can keep the value of the interval in a state and change it to stop the interval

**Comment by vivekimsit** [comment]
Thanks for the quick reply. I tried your approach and I am able to change the interval state but it seems that there is no effect on already setup request hook.

```js
const [interval, setInterval] = useState(0);
const {data} = myCustomHook(resourceId, {refershInterval: interval, revalidateOnFocus: false});

useEffect((
  if (shouldRefetch(car)) {
    setInterval(5000);
  } else {
    setInterval(0);
  }
) => {}, [car]);
```

**Comment by vivekimsit** [comment]
Ok, so it works fine while setting `5000` from `0` but not opposite.

**Comment by shuding** [comment]
Thank you @vivekimsit! Mind creating a repro on codesandbox.io?

**Comment by gone** [comment]
I am seeing a similar issue with using state to track revalidateOnFocus - it configures correctly to start but doesn't change as state updates.

**Comment by shuding** [comment]
Closing as this should have been fixed in #638.
